doc/kernel/kernel.rst

```diff
@@ -25,10 +25,25 @@ include: fitness wearables, smart watches, and IoT wireless gateways.
 .. toctree::
    :maxdepth: 2
 
-   threads/threads.rst
-   timing/timing.rst
-   memory/memory.rst
-   synchronization/synchronization.rst
-   data_passing/data_passing.rst
+   threads/index.rst
```

Comment:
If you wanted to group these with a heading for each group, you can use multiple toctree directives...